src/solvers/mp/RotationMP.cpp

```diff
@@ -0,0 +1,928 @@
+//
+// Created by antoine legrain on 2020-04-15.
+//
+
+#include "RotationMP.h"
+#include "solvers/mp/modeler/BcpModeler.h"
+#include "solvers/mp/RCPricer.h"
+#include "solvers/mp/TreeManager.h"
+
+using std::vector;
+using std::map;
+using std::pair;
+using std::min;
+using std::max;
+using std::string;
+using std::cout;
+using std::endl;
+
+//-----------------------------------------------------------------------------
+//
+//  S t r u c t   R o t a t i o n
+//
+//  A rotation is a set of shifts for a set of consecutive days.
+//  It has a cost and a dual cost (tbd).
+//
+//-----------------------------------------------------------------------------
+
+void Rotation::computeCost(Scenario* pScenario, const vector<LiveNurse*>& liveNurses, int horizon){
```

Comment:
One more thought that came while reading this. Do we want to add a (long-term) issue to have some more general representation of constraints? It seems we have: 
- individual global constraints on sums of patterns (working days, week-ends) where we could add number of nights, etc. if needed
- individual constraints on successions of patterns
- individual forbidden patterns (forbidden successions, but also shift-off preferences maybe)
- constraints on groups of nurses
 If  we're able to do such factoring, it may make future evolutions easier...

Reply:
i think it would be nice to move to a more general representation of the constraints. Long term issue for sure !